packages/shortest/src/core/test-generator/index.ts

```diff
@@ -0,0 +1,157 @@
+import { DOT_SHORTEST_DIR_PATH } from "@/cache";
+import { SHORTEST_NAME } from "@/cli/commands/shortest";
+import { formatCode } from "@/core/test-generator/utils/format-code";
+import { lintCode } from "@/core/test-generator/utils/lint-code";
+import { getLogger } from "@/log";
+import { getErrorDetails } from "@/utils/errors";
+
+export const SHORTEST_DIR_NAME = "shortest";
+export const SHORTEST_DIR_PATH = path.join(process.cwd(), SHORTEST_DIR_NAME);
+const SHORTEST_EXPECT_NAME = "expect";
+
+const require = createRequire(import.meta.url);
+const generate = require("@babel/generator").default;
+
+export class TestGenerator {
+  private rootDir: string;
+  private framework: string;
+  private log = getLogger();
+
+  private readonly outputPath: string;
+  private readonly TEST_FILE_NAME = "functional.test.ts";
+  private readonly frameworkDir: string;
+
+  constructor(rootDir: string, framework: string) {
+    this.rootDir = rootDir;
+    this.framework = framework;
+    this.frameworkDir = path.join(DOT_SHORTEST_DIR_PATH, this.framework);
+    this.outputPath = path.join(SHORTEST_DIR_PATH, this.TEST_FILE_NAME);
+  }
+
+  public async execute(options: { force?: boolean } = {}): Promise<void> {
+    this.log.trace("Generating tests...", { framework: this.framework });
+
+    if (!options.force) {
+      if (await this.testFileExists()) {
+        this.log.trace("Test file already exists, skipping generation", {
+          path: this.outputPath,
+        });
+        return;
+      }
+    }
+
+    await this.generateTestFile();
+  }
+
+  private async testFileExists(): Promise<boolean> {
+    try {
+      await fs.access(this.outputPath);
+      return true;
+    } catch {
+      return false;
+    }
+  }
+
+  private async generateTestFile(): Promise<void> {
+    const rawFileContent = await this.generateRawFileOutput();
+    const formattedCode = await formatCode(rawFileContent, this.rootDir);
+    const lintedCode = await lintCode(formattedCode, this.rootDir);
+
+    try {
+      await fs.mkdir(SHORTEST_DIR_PATH, { recursive: true });
+      await fs.writeFile(this.outputPath, lintedCode);
+      this.log.info("Test file generated successfully", {
+        path: this.outputPath,
+      });
+    } catch (error) {
+      this.log.error("Failed to write tests to file", getErrorDetails(error));
+      throw error;
+    }
+  }
+
+  private async generateRawFileOutput(): Promise<string> {
+    const testPlans = await this.getTestPlans();
+
+    const importStatement = t.importDeclaration(
+      [
+        t.importSpecifier(
+          t.identifier(SHORTEST_NAME),
+          t.identifier(SHORTEST_NAME),
+        ),
+      ],
+      t.stringLiteral("@antiwork/shortest"),
+    );
+
+    const testStatements = testPlans
+      .map((plan) => {
+        const statements: t.Statement[] = [];
+
+        const statementArgs: any[] = [t.stringLiteral(plan.steps[0])];
+
+        if (plan.options?.requiresAuth) {
```

Comment:
Simple implementation to support tests that require authentication. The ENVs can be supplied during the initial setup (that will be interactive).